internal/pkg/agent/cmd/container.go

```diff
@@ -40,9 +40,9 @@ import (
 const (
 	requestRetrySleepEnv     = "KIBANA_REQUEST_RETRY_SLEEP"
 	maxRequestRetriesEnv     = "KIBANA_REQUEST_RETRY_COUNT"
-	defaultRequestRetrySleep = "1s"                             // sleep 1 sec between retries for HTTP requests
-	defaultMaxRequestRetries = "30"                             // maximum number of retries for HTTP requests
-	defaultStateDirectory    = "/usr/share/elastic-agent/state" // directory that will hold the state data
```

Comment:
this is here probably to share state across runs. have mounted volume. 
this PR fixes the problem but i don't believe it's proper fix. 

i believe proper fix would be to fix not just `data/downloads` but also `data/components` 
will prepare PR on monday and test it as well